src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -282,8 +284,9 @@ export class MeetingManager implements AudioVideoObserver {
     // This results into missing few meeting events triggered with audioVideoDidStop such as meetingEnded, meetingFailed and meetingStartFailed.
     // We may also loose audioInputUnselected and videoInputUnselected events as we choose null devices when an attendee leaves.
     // When a new AudioVideoFacade object is created remove and re-add the eventDidReceive observer which wont leak.
-    this.audioVideo.removeObserver({ eventDidReceive: this.eventDidReceive });
-    this.audioVideo.addObserver({ eventDidReceive: this.eventDidReceive });
+    this.eventDidReceiveObserver && this.audioVideo.removeObserver(this.eventDidReceiveObserver);
```

Comment:
This code is confusing. Why do we need to add several audioVideo observer here?
Can we just have one?

Reply:
The audioVideoDidStart and audioVideoDidStop were kept separate due to the meeting events issue.

Reply:
I think we don't need this condition check here, but maybe it would be helpful for readability.
According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/delete, delete an item that is not in the Set won't trigger any error, it will just return `false`

Reply:
I think we don't need this condition check here.
According to [MDN/set/delete](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/delete#return_value), if we delete an item that is not in the Set, it won't throw any error, just return `false`.

Reply:
I removed the part to down setupAudioVideoObservers section to keep addition, removal in one place. But this still needs to be separate from the existing observers and the comment still applies tested it.